### DISCUSSION TITLE
How to make `refetchQueries` trigger even when the cache is empty?

### DISCUSSION THREAD

**Discussion by muh-osman** [TanStack/query#7517] [Q&A]
I have two web pages: one that displays all `products` and another that allows you to `upload` a new product.

Usually, `refetchQueries` are triggered automatically when I add a new product, but this only happens if the products were cached previously (meaning the products page was visited before the upload page).

However, if I upload a new product without first visiting the ‘All Products’ page (meaning no data is cached), `refetchQueries` does not trigger.

> **My question is**: How can I fetch all products after uploading a new product, **regardless** of whether [“products”] were fetched before or not?

```js
import { useMutation, useQueryClient } from "@tanstack/react-query";
import api from "./Api";

export const useUploadProduct = () => {
  const qc = useQueryClient();

  return useMutation({
    mutationFn: async (newProduct) => {
      const res = await api.post("/products", newProduct);
      return res.data;
    },

    onSuccess: () => {
      qc.refetchQueries(["products"]);
    },
  });
};
```

**Comment by TkDodo** [comment]
with `queryClient.fetchQuery` or `queryClient.prefetchQuery`
